### PR TITLE
feat: support sending sync requests concurrently in batches when putting vmss vm

### PR DESCRIPTION
**Pull request by nilo19**
#### What type of PR is this?

/kind feature

#### What this PR does / why we need it:

Introduce a configuration option `putVMSSVMBatchSize`. If set, the sync requests will be sent concurrently in batches when putting vmss vms.

#### Which issue(s) this PR fixes:

Fixes #

#### Does this PR introduce a user-facing change?

```release-note
Introduce a configuration option `putVMSSVMBatchSize`. If set, the sync requests will be sent concurrently in batches when putting vmss vms.
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```

### CODE REVIEW

**Review by feiskyer — APPROVED** [review]
/lgtm

After merged, please open new PRs for 1) update the docs and 2) cherry pick changes to release branches.

File: pkg/azureclients/armclient/azure_armclient.go [hunk 1]
```diff
@@ -478,22 +439,111 @@ func (c *Client) PutResources(ctx context.Context, resources map[string]interfac
 				}
 
 				responseLock.Lock()
-				responses[resourceID] = &PutResourcesResponse{
+				previousResponses[resourceID] = &PutResourcesResponse{
```
Comment (MartinForReal) [inline pkg/azureclients/armclient/azure_armclient.go:442]: I'm wondering if sync.Map would be a fit for this scenario.

File: pkg/azureclients/vmssvmclient/azure_vmssvmclient.go [hunk 2]
```diff
@@ -470,7 +472,7 @@ func (c *Client) UpdateVMs(ctx context.Context, resourceGroupName string, VMScal
 }
 
 // updateVMSSVMs updates a list of VirtualMachineScaleSetVM from map[instanceID]compute.VirtualMachineScaleSetVM.
-func (c *Client) updateVMSSVMs(ctx context.Context, resourceGroupName string, VMScaleSetName string, instances map[string]compute.VirtualMachineScaleSetVM) *retry.Error {
+func (c *Client) updateVMSSVMs(ctx context.Context, resourceGroupName string, VMScaleSetName string, instances map[string]compute.VirtualMachineScaleSetVM, batchSize int) *retry.Error {
```
Comment (MartinForReal) [inline pkg/azureclients/vmssvmclient/azure_vmssvmclient.go:475]: I'm wondering if we should add another function instead to keep backward compatible.  However, There is no other usage. I think it is fine.

**Review by MartinForReal — CHANGES_REQUESTED** [review]

**Pull request merged**